### DISCUSSION TITLE
[@types/xrm] missing types in control interface

### DISCUSSION THREAD

**Discussion by VictorSijtsma** [DefinitelyTyped/DefinitelyTyped#56180] [Issues with a @types package] [answered]
types/xrm
I added for myself the last 3 lines. Not sure if they are completely perfect..
Don't know if it is handy to add these to the index.d.ts

links to the microsoft docs:
https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/formcontext-ui/clearformnotification
https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/controls/setvisible
https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/controls/setdisabled

    /**
         * Interface for controls.
         *
         * @see {@link UiElement}
         */
        interface Control extends UiLabelElement, UiCanGetVisibleElement {
            /**
             * Gets the controls type.
             * @returns The control type.<BR><BR>
             * **Values returned are**:
             * * standard
             * * iframe
             * * lookup
             * * optionset
             * * subgrid
             * * webresource
             * * notes
             * * timercontrol
             * * kbsearch (CRM Online Only, use parature.d.ts)
             * * quickform (see ui.QuickForm)
             * * customcontrol: <namespace>.<name> (A custom control for mobile phone and tablet clients).
             * * customsubgrid: <namespace>.<name> (A custom dataset control for mobile phone and tablet clients).
             */
            getControlType(): ControlType | string;

            /**
             * Gets the name of the control on the form.
             * @returns The name of the control.
             * @remarks     The name assigned to a control is not determined until the form loads. Changes to
             *              the form may change the name assigned to a given control.
             *              When you use the control getName method the name of the first control will be the
             *              same as the name of the attribute. The second instance of a control for that
             *              attribute will be "<attributeName>1". The pattern <attributeName>+N
             *              will continue for each additional control added to the form for a specific
             *              attribute. When a form displays a business process flow control in the header,
             *              additional controls will be added for each attribute that is displayed in the
             *              business process flow. These controls have a unique name like the following:
             *              header_process_<attribute name>.
             */
            getName(): string;

            /**
             * Gets a reference to the Section parent of the control.
             * @returns The parent Section.
             */
            getParent(): Section;

            **clearNotification(): boolean; // new

            setVisible(bool): void; // new

            setDisabled(bool): void; // new**
        }

[a comment by an automated account is not shown]

**Comment by daryllabar** [comment, marked as answer]
The clearNotification doc you list is not on the control, but on the form itself.  AS for the other functions, they are on the StandardControl.  That is probably the type you want to be using.

**Reply by VictorSijtsma** [reply]
Hi, thanks for your quick response. I see them, but those specifically are not found by the typing. 
I meant this one: https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/controls/clearnotification
which is also on the standard control, but is also not found by the typing, for some reason.. :(
sorry, found it, i Typed Xrm.Controls.Control, but it has to be (like you proposed) Xrm.Control.StandardControl.

Thank A lot :D

**Reply by daryllabar** [reply]
The standard getControl is typed to return the Control interface.  You'll have to cast it to the correct Control type
